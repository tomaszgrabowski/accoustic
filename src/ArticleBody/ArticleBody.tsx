import React from 'react';
import Section from '../components/Section/Section';

interface IProps {
    data: string[]
}

const ArticleBody = ( props: IProps ) => {
    return (
        <main>
            { props.data.map( value => <Section key={ Math.random() } data={ value }/> ) }
        </main>
    );
};

export default ArticleBody;
