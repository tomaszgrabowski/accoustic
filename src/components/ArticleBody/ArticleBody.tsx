import React from 'react';
import Section from '../Section/Section';
import './ArticleBody.css'

interface IProps {
    data: string[]
}

const ArticleBody = ( props: IProps ) => {
    return (
        <main>
            { props.data.map( value => <Section key={ Math.random() } data={ value }/> ) }
        </main>
    );
};

export default ArticleBody;
