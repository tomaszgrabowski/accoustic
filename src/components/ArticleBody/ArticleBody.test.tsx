import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import ArticleBody from './ArticleBody';

describe( 'ArticleBody', () => {
    let testData: string[] = [];
    
    beforeEach( () => testData = ['a', 'b', 'c'] );
    
    test( 'given data (array of strings) when render then show list of sections', () => {
        const wrapper = shallow( <ArticleBody data={ testData }/> );
        expect( wrapper.exists( 'main' ) ).toBeTruthy();
        expect( wrapper.find( 'main' ).children().length ).toEqual( 3 );
    } );
    test( 'given empty data when render then show empty list', () => {
        const data = [] as string[];
        const wrapper = shallow( <ArticleBody data={ data }/> );
        expect( wrapper.exists( 'main' ) ).toBeTruthy();
        expect( wrapper.find( 'main' ).children().length ).toEqual( 0 );
    } );
    
    test( 'given ArticleBody component when render then renders properly', () => {
        const wrapper = renderer.create( <ArticleBody data={ testData }/> ).toJSON();
        expect( wrapper ).toMatchSnapshot();
    } );
} );
