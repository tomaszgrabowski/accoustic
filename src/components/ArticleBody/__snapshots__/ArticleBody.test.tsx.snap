// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleBody given ArticleBody component when render then renders properly 1`] = `
<main>
  <section
    dangerouslySetInnerHTML={
      Object {
        "__html": "a",
      }
    }
  />
  <section
    dangerouslySetInnerHTML={
      Object {
        "__html": "b",
      }
    }
  />
  <section
    dangerouslySetInnerHTML={
      Object {
        "__html": "c",
      }
    }
  />
</main>
`;
